Source/JavaScriptCore/jit/AssemblyHelpers.h

```diff
@@ -1962,6 +1962,47 @@ class AssemblyHelpers : public MacroAssembler {
     static bool storeWasmContextInstanceNeedsMacroScratchRegister();
 #endif
 
+    void emitFillStorageWithJSEmpty(GPRReg baseGPR, ptrdiff_t initialOffset, unsigned count, GPRReg scratchGPR)
+    {
+#if USE(JSVALUE64)
+        unsigned pairCount = count >> 1;
+        unsigned pairIndex = 0;
+        ASSERT(JSValue::encode(JSValue()) == 0);
+#if CPU(ARM64)
+        UNUSED_PARAM(scratchGPR);
+        GPRReg emptyValueGPR = ARM64Registers::zr;
+#else
+        GPRReg emptyValueGPR = scratchGPR;
+        move(TrustedImm32(0), scratchGPR);
+#endif
+        for (; pairIndex < pairCount; ++pairIndex)
+            storePair64(emptyValueGPR, emptyValueGPR, baseGPR, TrustedImm32(initialOffset + pairIndex * 2 * sizeof(EncodedJSValue)));
```

Comment:
While this can work, if the offset ends up being > 128, then this `storePair64` will downgrade to 2 `store64`.  Do we know what the typical `count` is and how often it gets big?  If we can afford another scratch registers to use as the base register or if the baseGPR can be clobbered, the other way to do this is to use the spooler which will adjust the baseGPR as needed.  Alternatively, the underlying ARM64 `stp` instruction supports post increment mode, thought we have not exposed that in the MacroAssembler yet.  If we want to, we can add and use a `storePairPostIncrement64` instead to guarantee that we'll never exceed the 128 offset limit.

Reply:
The count would be smaller. But even if it is wider than that, then still, it is just the same to the old implementation (store64). So this is fine.